src/main/java/org/mbari/cthulhu/ui/player/MediaPlayerTimer.java

```diff
@@ -51,48 +51,77 @@
-     * @param onTick function to execute on each tick of the timer depending of time of previous call
+     * @param onTick function to execute on each tick of the timer
      */
     MediaPlayerTimer(MediaPlayer mediaPlayer, Consumer<Long> onTick) {
         timer.scheduleAtFixedRate(new TimerTask() {
             @Override
             public void run() {
+                final long newMpMillis = mediaPlayer.status().time();
+                
                 if (mediaPlayer.status().isPlaying()) {
-                    final long newMpMillis = mediaPlayer.status().time();
+                    wasPlaying = true;
+                    
+                    final long currSysMillis = System.currentTimeMillis();
                     
                     // the value to use for onTick.accept
                     long toCallTickMillis;
-    
+
                     if (newMpMillis != previousMpMillis) {
                         // mediaPlayer has reported a new value.
                         timeDirection = newMpMillis >= previousMpMillis ? +1 : -1;
                         toCallTickMillis = previousMpMillis = newMpMillis;
-                        previousSysMillis = System.currentTimeMillis();
+                        previousSysMillisForMp = currSysMillis;
                     }
                     else {
                         // mediaPlayer has reported the same value as before.
                         // Let's use last known time direction and system time difference for the onTick call:
                         toCallTickMillis = previousMpMillis +
-                            timeDirection * (System.currentTimeMillis() - previousSysMillis);
+                            timeDirection * (currSysMillis - previousSysMillisForMp);
                     }
                     
-                    log.trace("onTick: toCallTickMillis={}  timeDirection={}", toCallTickMillis, timeDirection);
+                    if (log.isTraceEnabled()) {
+                        final long sysDiffMillis = currSysMillis - previousSysMillis;
+                        previousSysMillis = currSysMillis;
+                        log.trace("onTick: toCallTickMillis={} newMpMillis={} timeDirection={} (sysDiff={})",
+                            toCallTickMillis, newMpMillis, timeDirection, sysDiffMillis
+                        );
+                    }
+
                     onTick.accept(toCallTickMillis);
                 }
                 else {
                     previousMpMillis = -1;
+                    if (wasPlaying) {
+                        // Media just paused. Let's do one more call to the handler indicating
+                        // the time reported by the media player:
+                        wasPlaying = false;
+                        log.trace("onTick: media just paused. Calling handler with newMpMillis={}", newMpMillis);
+                        onTick.accept(newMpMillis);
```

Comment:
Note to myself: This is the major change here ... updating the `Consumer<Long> onTick` when paused.